erc/ERCS/erc-7562.md

```diff
@@ -180,42 +272,42 @@ To help make sense of these params, note that a malicious paymaster can at most
     * **[OP-053]** May call the fallback function from the `sender` with any value
     * **[OP-054]** Any other access to the `EntryPoint` is forbidden
 * `*CALL` opcodes:
     * **[OP-061]** `CALL` with `value` is forbidden. The only exception is a call to the `EntryPoint` described above.
     * **[OP-062]** Precompiles:
         * Only allowed the core 9 precompiles.\
-          Specifically the computation precompiles that do not access anything in the blockchain state or environment.
-* **[OP-070] Transient Storage slots defined in [EIP-1153](./eip-1153.md) and accessed using `TLOAD` (`0x5c`) and `TSTORE` (`0x5d`) opcodes
-    are treated exactly like persistent storage (SLOAD/SSTORE)
+          Specifically, the computation precompiles that do not access anything in the blockchain state or environment.
+* **[OP-070]** Transient Storage slots defined in [EIP-1153](./eip-1153) and accessed using `TLOAD` (`0x5c`) and `TSTORE` (`0x5d`) opcodes
+  are treated exactly like persistent storage (SLOAD/SSTORE)
 
 ### Code Rules
 
 * **[COD-010]** Between the first and the second validations, the `EXTCODEHASH` value of any visited address,
-entity or referenced library, may not be changed.\
-If the code is modified, the UserOperation is considered invalid.
+  entity, or referenced library, may not be changed.\
+  If the code is modified, the UserOperation is considered invalid.
 
 ### Storage rules.
 
-The permanent storage access with `SLOAD` and `SSTORE` instructions within each phase are limited as follows:
+The permanent storage access with `SLOAD` and `SSTORE` instructions within each phase is limited as follows:
 
 * **[STO-010]** Access to the "account" storage is always allowed.
 * Access to associated storage of the account in an external (non-entity contract) is allowed if either:
     * **[STO-021]**  The account already exists
     * **[STO-022]**  There is an `initCode` and the `factory` contract is staked.
 * If the entity (`paymaster`, `factory`) is staked, then it is also allowed:
     * **[STO-031]** Access the entity's own storage.
-    * **[STO-032]** Read/Write Access to storage slots that is associated with the entity, in any non-entity contract.
+    * **[STO-032]** Read/Write Access to storage slots that are associated with the entity, in any non-entity contract.
     * **[STO-033]** Read-only access to any storage in non-entity contract.
 * **[STO-040]** `UserOperation` may not use an entity address (`factory`/`paymaster`/`aggregator`) that is used as an "account" in another `UserOperation` in the mempool. \
-This means that a `Paymaster` and `Factory` contracts cannot practically be an "account" contract as well.
+  This means that `Paymaster` and `Factory` contracts cannot practically be an "account" contract as well.
 * **[STO-041]** A contract whose Associated Storage slot is accessed during a `UserOperation` may not be an address of a "sender" in another `UserOperation` in the mempool.
```

Comment:
rephrased.

Reply:
I'm looking at the current wording and I still don't read it as what we intended to do:

`* **[STO-041]** A contract whose storage is associated with any entity of a `UserOperation` may not be a "sender" of another UserOperation in the mempool.`

It still sounds to me like "if you have two UserOperations, op1 and op2, where op1.sender is accessed by op2, exclude **op1**".  Whereas the intended behavior is the opposite: "if you have two UserOperations, op1 and op2, where op1.sender is accessed by op2, exclude **op2**"

Isn't that the intended behavior?

And also, does it need to be "in the mempool" rather than "in the same bundle", or is that just an implementation detail in our own bundler?